ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -920,3 +923,322 @@ def test_ipa_migrate_stage_mode_with_invalid_cert(self):
+            "  First RID of the corresponding RID range: 300000\n"
+            "  First RID of the secondary RID range: 400000\n"
+            "  Range type: local domain range\n"
+            "----------------------------\n"
+            "Number of entries returned 1\n"
+            "----------------------------\n"
+        )
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
+            ["ipa", "idrange-find", range_name])
+        assert CMD_OUTPUT in cmd.stdout_text
+
+    def test_automountlocation_is_migrated(self):
+        """
+        This testcase checks that automount location/maps
+        and keys are migrated.
+        """
+        base_dn = str(self.master.domain.basedn)
+        automount_cn = "automount"
+        loc_name = "baltimore"
+        auto_map_name = "auto.share"
+        DEBUG_LOG = (
+            "Added entry: cn={},cn={},{}\n"
+        ).format(loc_name, automount_cn, base_dn)
+        CMD1_OUTPUT = (
+            "  Location: baltimore\n"
+        )
+        CMD2_OUTPUT = (
+            "  Map: auto.share\n"
+        )
+        CMD3_OUTPUT = (
+            "-----------------------\n"
+            "1 automount key matched\n"
+            "-----------------------\n"
+            "  Key: sub\n"
+            "  Mount information: -fstype=autofs ldap:auto.man\n"
+        )
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "automountlocation-show", loc_name])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "automountmap-find", loc_name])
+        cmd3 = self.replicas[0].run_command(
+            ["ipa", "automountkey-find", loc_name, auto_map_name]
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert CMD1_OUTPUT in cmd1.stdout_text
+        assert CMD2_OUTPUT in cmd2.stdout_text
+        assert CMD3_OUTPUT in cmd3.stdout_text
+        assert DEBUG_LOG in install_msg
+
+    def test_vault_is_migrated(self):
+        """
+        This testcase checks that vault is
+        not migrated
+        """
+        vault_name = "testvault"
+        CMD_OUTPUT = "1 vaults matched"
+        run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n']
+        )
+        cmd = self.replicas[0].run_command(
+            ["ipa", "vault-find", vault_name])
```

Comment:
You need to add `raiseonerr=False` to the call to `run_command`, then `assert cmd.returncode != 0` and `assert "0 vaults matched" in cmd.stdout_text`